### PR TITLE
[15.0][FIX] crm_multicompany_reporting_currency: Set currency_rate if company isn't specified on lead and correct currency_rate calculation

### PR DESCRIPTION
**Pull request by yankinmax**

### CODE REVIEW

File: crm_multicompany_reporting_currency/models/crm_lead.py [hunk 1]
```diff
@@ -51,14 +51,20 @@ def _compute_multicompany_reporting_currency_id(self):
     def _compute_currency_rate(self):
         # similar to currency_rate on sale.order
         for record in self:
-            if (
+            date = record.create_date or fields.Date.today()
+            if not record.company_id:
+                record.currency_rate = (
+                    record.company_currency.with_context(date=date).rate or 1.0
+                )
+                continue
```
Comment (jcoux) [inline crm_multicompany_reporting_currency/models/crm_lead.py]: `continue` is useless as rest of the function is under condition `elif` and `else`.
```suggestion
```

**Review by jcoux — COMMENTED** [review]
Little suggestion

**Review by jcoux — APPROVED** [review]

**Review by ajaniszewska-dev — APPROVED** [review]

**Comment by yankinmax** [comment]
@OCA/community-maintainers can we merge this fix, please?

**Review by ivantodorovich — APPROVED** [review]

**Comment by ivantodorovich** [comment]
@yankinmax , I recommend you add yourself as maintainer of the module, so you can merge fixes yourself ;)

Of course it won't work for this PR (as you're not a maintainer now), but it'll work for future PRs

**Comment by yankinmax** [comment]
@gurneyalex @rousseldenis can you pls help to get this merged?

**Review by gurneyalex — APPROVED** [review]

**Pull request merged**